[stable/concourse] promote pod affinity from annotation to spec

erickt opened the pull request:
This updates @pfeodrippe's pull request to address
@unguiculus's comments. It moves the affinity into the
kubernetes 1.7 compatible affinity. It follows the pattern
used to fix cockroachdb in #1326.

Closes #1462, #1525, #1306

pfeodrippe committed:
[stable/concourse] fix json to work with kubernetes v1.7

This updates @pfeodrippe's pull request to address
@unguiculus's comments. It moves the affinity into the
kubernetes 1.7 compatible affinity. It follows the pattern
used to fix cockroachdb in #1326.

Closes #1462, #1525, #1306

dhilipkumars commented on stable/concourse/templates/worker-statefulset.yaml:
```diff
@@ -90,6 +71,18 @@ spec:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
```
do you want to swtich to `matchLabels` instead of `labelSelector` the former is easier to use. 
```
labelSelector:
              matchLabels:
                app:  {{ template "fullname" . }}
                release: {{ .Release.Name | quote }}
```

dhilipkumars replied:
[ref](https://github.com/kubernetes/charts/blob/master/stable/memcached/templates/statefulset.yaml)

dhilipkumars commented on stable/concourse/templates/worker-statefulset.yaml:
```diff
@@ -90,6 +71,18 @@ spec:
+                  operator: In
+                  values:
+                  - {{ template "worker.fullname" . }}
+              topologyKey: kubernetes.io/hostname
```
could you move 
```
topologyKey: kubernetes.io/hostname
```
like this?
```
podAffinityTerm:
            - topologyKey: "kubernetes.io/hostname"
              labelSelector:
```

gives better readablity.

dhilipkumars commented:
/ok-to-test

dhilipkumars commented:
/assign

dhilipkumars commented:
could you also update the `README.md` about the prerequisite?

erickt committed:
[stable/concourse] Add worker statefulset annotations

This allows us to attach things like kube2iam annotations to
the stateful sets to allow Concourse workers to access private ECR
repositories.

erickt committed:
[stable/concourse] Bump chart version

erickt commented:
@dhilipkumars: I believe I've addressed your suggestions. I also added a new feature, where we can add worker annotations. I need that to give the workers kube2iam permissions. If you want I can factor that out into a separate pull request. I also gave this a major bump to 0.4.0, so that people on kubernetes 1.5 clusters can continue to use the 0.3.x versions.

dhilipkumars commented:
@erickt Thanks for addressing the comments.
/test all

dhilipkumars commented:
/lgtm

pfeodrippe commented:
@erickt  Thanks, man! o/

erickt commented:
@pfeodrippe happy to help get it over the finish line!
